src/puppetlabs/trapperkeeper/internal.clj

```diff
@@ -210,15 +224,22 @@
   (async/go (try
               (loop []
                 (let [task (async/<! lifecycle-chan)]
-                  (if (= task :shutdown)
-                    (do
-                      (log/debug "Received shutdown command, lifecycle worker exiting.")
-                      :done)
-                    (do
-                      (log/debug "Lifecycle worker executing lifecycle task.")
-                      (task)
-                      (log/debug "Lifecycle worker completed lifecycle task; awaiting next task.")
-                      (recur)))))
+                  (schema/validate LifeCycleTask task)
+                  (let [{:keys [type task-function]} task]
+                    (condp #(contains? %1 %2) type
+                      #{:shutdown}
+                      (do
+                        (log/debug "Received shutdown command, lifecycle worker exiting.")
+                        :done)
+
+                      #{:boot :restart}
+                      (do
+                        (log/debugf "Lifecycle worker executing %s lifecycle task." type)
+                        (task-function)
```

Comment:
It looks like the schema would allow for the `task-function` to be optional.  If that's intentional, should we check that `task-function` has a non nil value before executing it?  Assuming `task-function` should be required for a `:boot` or `:restart` task, should we harden the schema to ensure that it has a non nil value?